go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/celestiaorg/celestia-app
 
-go 1.21.1
+go 1.22
+
+toolchain go1.22.0
```

Comment:
ideally we would do this in a separate PR, is this required for the other changes here? if so, then we will likely need to create a different PR that also updates all the CI to use the latest go version